browser/mapping.go

```diff
@@ -457,6 +465,39 @@ func mapFrame(vu moduleVU, f *common.Frame) mapping {
 	return maps
 }
 
+func parseWaitForFunctionArgs(
+	ctx context.Context, timeout time.Duration, pageFunc, opts goja.Value, args ...goja.Value,
+) (string, *common.FrameWaitForFunctionOptions, []any, error) {
+	popts := common.NewFrameWaitForFunctionOptions(timeout)
+	err := popts.Parse(ctx, opts)
+	if err != nil {
+		return "", nil, nil, fmt.Errorf("parsing waitForFunction options: %w", err)
+	}
+
+	js := pageFunc.ToString().String()
+	_, isCallable := goja.AssertFunction(pageFunc)
+	if !isCallable {
+		js = fmt.Sprintf("() => (%s)", js)
+	}
+
+	pargs := make([]any, 0, len(args))
+	for _, arg := range args {
+		pargs = append(pargs, arg.Export())
+	}
+
+	return js, popts, pargs, nil
+}
+
+// MapPage will map a page for an integration test. It is only to be used for
+// internal testing purposes.
+func MapPage(vu k6modules.VU, p *common.Page) mapping {
```

Comment:
Not sure about this either. It's only meant to for integration tests. WDYT?

Reply:
A new method of mapping the desired types was used, which was already being used by `TestTracing`. See 80598a1f7fd47e91fc340fbdb0a589f68618e4b0, 97420ece4f36426df6932d509d7ace38d8a516e4 and 6c957279b9f08c986b9554d4e242af714b6ccfb3.